File: AtCoder/ABC/ABC017/B.java
import java.util.*;
import java.io.*;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    String s = sc.next();
    int i = 0;
    boolean condition = true;
    while (i < s.length()) {
      if (s.charAt(i) == 'o' || s.charAt(i) == 'k' || s.charAt(i) == 'u')
        i++;
      else if (s.charAt(i) == 'c' && s.charAt(i + 1) == 'h')
        i += 2;
      else {
        condition = false;
        break;
      }
    }
    System.out.println(condition ? "YES" : "NO");
  }
}

File: AtCoder/ABC/ABC017/B.cc
#include <iostream>
#include <string>

using namespace std;

int main()
{
  int i = 0;
  string s;
  bool condition = true;
  cin >> s;
  while (i < s.size())
  {
    if (s.at(i) == 'o' || s.at(i) == 'k' || s.at(i) == 'u')
      i++;
    else if (s.at(i) == 'c' && s.at(i + 1) == 'h')
      i += 2;
    else
    {
      condition = false;
      break;
    }
  }
  cout << (condition ? "YES" : "NO") << endl;
}
